src/NUnitFramework/benchmarks/nunit.framework.benchmarks/CompositeWorkItemBenchmark.cs

```diff
@@ -0,0 +1,107 @@
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using BenchmarkDotNet.Attributes;
+using NUnit.Framework.Api;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Abstractions;
+using NUnit.Framework.Internal.Execution;
+
+namespace NUnit.Framework;
+
+[MemoryDiagnoser]
+public class CompositeWorkItemBenchmark
+{
+    private Dictionary<string, object> _loaderOptions;
+
+    [GlobalSetup]
+    public void GlobalSetup()
+    {
+        _loaderOptions = new Dictionary<string, object>
+        {
+            { "ProcessModel", "InProcess" },
+            { "DomainUsage", "None" },
+            { "ShadowCopyFiles", false },
+            { "TestParametersDictionary", "" },
+            { "NumberOfTestWorkers", 0 },
+            { "SynchronousEvents", "false" },
+            { "RandomSeed", 878248866 },
+            { "LOAD", new List<string> { "NUnit.Framework.TestWithTestActionAttribute" } },
+            { "WorkDirectory", Path.GetTempPath() },
+        };
+    }
+
+    [Benchmark]
+    public void TestActionUsage()
+    {
+        DefaultTestAssemblyBuilder builder = new DefaultTestAssemblyBuilder();
+        var test = builder.Build(typeof(TestWithTestActionAttribute).Assembly, _loaderOptions);
+        WorkItem workItem = WorkItemBuilder.CreateWorkItem(test, TestFilter.Empty, new DebuggerProxy(), recursive: true);
+        var context = new TestExecutionContext();
+        context.Dispatcher = new SuperSimpleDispatcher();
+        workItem.InitializeContext(context);
+        workItem.Execute();
+    }
+}
+
+[SomeTestAttr]
+[TestFixtureSource(nameof(_fixtureArgs))]
+public class TestWithTestActionAttribute
+{
+    private static object[] _fixtureArgs = Enumerable.Range(0, 100)
+                                                     .Select(a => new object[] { $"a{a:000}" })
+                                                     .ToArray();
+
+    public TestWithTestActionAttribute(string arg) { }
+
+    [Test]
+    public void Test1([Range(0, 10)] int x) { }
+}
+
+[AttributeUsage(AttributeTargets.Class)]
+public class SomeTestAttrAttribute : Attribute, ITestAction
+{
+    private object _data = Enumerable.Range(0, 1000000).ToArray();
```

Comment:
I presume this is to make this a slow attribute in order to see the difference in caching.